source/administration/update_mattermost.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+##Configuration - please adapt it to your environment
+#path to mattermost directory
+mattermostdir=/opt/mattermost
+#path for backup
+backupdir=/opt
+#temporary path for download
+downloaddir=/tmp
+#Team or Enterprise edition? Set 1 for team edition
+teamedition=0
+#start with plugins? Set 1 for starting with plugins active
+plugins=0
+#########
+
+##set version number from argument
+version=$1
+
+#set variable for echo output
+if [ $teamedition -ne 0 ]; then
+        edition=Team
+        downloadlocation=https://releases.mattermost.com/$version/mattermost-team-$version-linux-amd64.tar.gz
+else
+        edition=Enterprise
+        downloadlocation=https://releases.mattermost.com/$version/mattermost-$version-linux-amd64.tar.gz
+fi
+
+#delete old backup folder
+if [ -d "$backupdir/mattermost-backup" ]; then
+        echo Deleting old mattermost backup
+        rm -rf $backupdir/mattermost-backup
+fi
+
+###Get the file
+echo Downloading Mattermost $edition $version
+wget $downloadlocation -P $downloaddir -q
+
+#Stop if download fails
+if [ $? -ne 0 ]; then
+        echo Mattermost download failed
+        exit 1
+fi
+echo Download successfull
+
+#unpack the tar file
+echo unpacking
+tar -x --transform='s,^[^/]\+,\0-upgrade,' -f $downloaddir/mattermost*.gz -C $downloaddir
+
+#stopping mattermost service
+echo Stopping Mattermost service
+if [ -f /bin/systemctl ];  then
+        servicecommand="systemctl stop mattermost"
+else
+        servicecommand="service mattermost stop"
+fi
+$servicecommand
+
+#check if mattermost is still running
+if pgrep mattermost > /dev/null
+then
+        echo Mattermost still running. Update not possible.
+        echo cleaning
+        rm -rf $downloaddir/mattermost-upgrade/
+        rm -f $downloaddir/mattermost*.gz
+        exit 1
+fi
+
+echo Create backup of mattermost directory
+cp -ra $mattermostdir/ $backupdir/mattermost-backup/
```

Comment:
Maybe align back up directory naming convention similar to manual where it includes date/time as name suffix?

Reply:
I made it fix as it simplified the script and the backup from 2 version ago doesn’t matter right? Because when you need to rollback the Binary and the DB you are anyway screwed because of loosing data

Reply:
Each version is different (folder might change, JS files/maps are different).  So if the admin, didn't clear up the back up directory before upgrade, it will end up with mixed folders/files from old and new version, and might not be usable in case the admin needs to revert from the recent upgrade. I think that's the reason why it ends up with backing up to new folder (with date/time) so that admin don't need to worry about that and may attend to those back up folders separately.

Reply:
? Thats the reason why the backup folder gets first deleted. So it the admin has always the recent backup.

Example:
- current version is 5.7.1
- admin uses the script to update to 5.8.0
- backup contains 5.7.1
- admin uses the script to update to 5.9.0
- backup of 5.7.1 get's deleted
- backup of 5.8.0 is created
- update to 5.9.0 happen

Reply:
Oh right, I missed that delete backup block above, maybe move it near this create block?
But still I'm in the impression that this "with script" and manual upgrades should at least be doing the same, especially when saving backup directory. Non-blocking, just need to give more thought about it.